### ISSUE TITLE
`nix-env` can't evaluate Nixpkgs with GCC 9

### ISSUE THREAD

**Issue by grahamc** [NixOS/nix#3300]
Nixpkgs just updated to GCC9 by default, in https://github.com/nixos/nixpkgs/commit/8f729c0070ec3f78edadeaebcbd110257fe4577e which is now in master. Before this commit, Nix can evaluate nixpkgs just fine -- but after ...

```
$ nix-build ./nixos/release-combined.nix -A nixpkgs.tarball --no-out-link -j1 --cores 8
warning: dumping very large path (> 256 MiB); this may run out of memory
these derivations will be built:
  /nix/store/xpkk55x6ni89nj0kxg6p2z3qx3hbv2pq-nixpkgs-lib-tests.drv
  /nix/store/pm0bka6jzgjjq1b1byagc5gvnzmk32m1-nixpkgs-tarball-20.03pre56789.gfedcba.drv
building '/nix/store/xpkk55x6ni89nj0kxg6p2z3qx3hbv2pq-nixpkgs-lib-tests.drv'...
====== module tests ======
68 Pass
0 Fail
building '/nix/store/pm0bka6jzgjjq1b1byagc5gvnzmk32m1-nixpkgs-tarball-20.03pre56789.gfedcba.drv'...
unpacking sources
unpacking source archive /nix/store/d4zy827k075333a4rpaqymn6gfsjxpmv-nixpkgs
source root is nixpkgs
patching sources
autoconfPhase
No bootstrap, bootstrap.sh, configure.in or configure.ac. Assuming this is not an GNU Autotools package.
configuring
release name is nixpkgs-20.03pre56789.gfedcba
git-revision is gfedcba
building
no Makefile, doing nothing
running tests
warning: you did not specify '--add-root'; the result might be removed by the garbage collector
warning: you did not specify '--add-root'; the result might be removed by the garbage collector
checking Nixpkgs on i686-linux
error: stack overflow (possible infinite recursion)
build time elapsed:  0m0.093s 0m0.038s 0m42.131s 0m5.349s
```

We are authoring a patch to pin Nix to gcc8Stdenv for now.

**Comment by edolstra** [comment]
This is probably fixed by 98ef11677c43db9aa669768d9f0cbec704e8831c.

**Comment by edolstra** [comment]
BTW the main culprit for stack overflows is `texlive.combined.scheme-full`, which does some very deep recursion.

**Comment by LnL7** [comment]
When tracing the eval I did indeed notice that the problem occurred somewhere in the texlive set. Hopefully we can revert the gcc8 changes once we have a release with this then.

**Comment by edolstra** [comment]
Fixed in cb90e382b5b6e177ea902b3909fd1897643ae3cd.
